titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/pod/KubePodUtil.java

```diff
@@ -247,6 +252,55 @@ private static String buildV1VolumeMountPropagation(String mountPropagation) {
         }
     }
 
+    public static List<V1Volume> buildV1Volumes(List<Volume> volumes) {
+        if (volumes == null) {
+            return Collections.emptyList();
+        }
+        List<V1Volume> v1Volumes = new ArrayList<>();
+        for (Volume v : volumes) {
+            buildV1Volume(v).ifPresent(v1Volumes::add);
+        }
+        return v1Volumes;
+    }
+
+    private static Optional<V1Volume> buildV1Volume(Volume volume) {
+        if (volume.getVolumeSource() instanceof SharedContainerVolumeSource) {
+            V1FlexVolumeSource flexVolume = getV1FlexVolumeForSharedContainerVolumeSource(volume);
+            return Optional.ofNullable(new V1Volume()
+                    .name(volume.getName())
+                    .flexVolume(flexVolume));
+        } else if (volume.getVolumeSource() instanceof SaaSVolumeSource) {
+            V1CephFSVolumeSource cephVolume = getV1CephVolumeSource(volume);
```

Comment:
In the future, if the SaaS volume ID can be backed by multiple storage providers, how does TJC know which one to configure here? Would it need to call the SaaS control plane to get its type, have the user provide it, or TBD?

Reply:
That is a titus-executor decision.
Ceph will probably still be the transport, but the backend store of the *ceph storage nodes* may be like, s3, ebs, ssds, who knows. 

If some day we decide to change the *transport*, TBD, but still will be an internal TJC decision.
Or heck maybe we'll just use/abuse the Ceph slot in the pod spec :)